modules/fluid_structure_interaction/src/interfacekernels/FluidStructureInterface.C

```diff
@@ -0,0 +1,160 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FluidStructureInterface.h"
+#include "MooseVariableFE.h"
+
+registerMooseObject("FluidStructureInteractionApp", FluidStructureInterface);
+
+InputParameters
+FluidStructureInterface::validParams()
+{
+  InputParameters params = InterfaceKernel::validParams();
+  params.addClassDescription("Enforces displacement and stress/pressure continuity "
+                             "between the fluid and structural domains. Element "
+                             "is always the structure and neighbor is always the"
+                             " fluid.");
+  params.addParam<MaterialPropertyName>("D", "D", "Fluid density.");
+  params.addRequiredParam<Real>("component", "The desired component of displacement.");
+  return params;
+}
+
+FluidStructureInterface::FluidStructureInterface(const InputParameters & parameters)
+  : InterfaceKernel(parameters),
+    _D(getMaterialProperty<Real>("D")),
+    _neighbor_dotdot(_neighbor_var.uDotDotNeighbor()),
+    _neighbor_dotdot_du(_neighbor_var.duDotDotDuNeighbor()),
+    _component(getParam<Real>("component"))
+{
+}
+
+Real
+FluidStructureInterface::computeQpResidual(Moose::DGResidualType type)
+{
+  Real r = 0;
+  if (_component == 0)
+  {
+
+    switch (type)
+    {
+
+      case Moose::Element: // Element is fluid
+        r = _test[_i][_qp] * _D[_qp] * _neighbor_dotdot[_qp] * _normals[_qp](0);
+        break;
+
+      case Moose::Neighbor: // Neighbor is structure
+        r = _test_neighbor[_i][_qp] * -_u[_qp] * _normals[_qp](0);
+        break;
+    }
+  }
+
+  if (_component == 1)
+  {
+    switch (type)
+    {
+
+      case Moose::Element: // Element is fluid
+        r = _test[_i][_qp] * _D[_qp] * _neighbor_dotdot[_qp] * _normals[_qp](1);
+        break;
+
+      case Moose::Neighbor: // Neighbor is structure
+        r = _test_neighbor[_i][_qp] * -_u[_qp] * _normals[_qp](1);
+        break;
+    }
+  }
+
+  if (_component == 2)
+  {
+    switch (type)
+    {
+
+      case Moose::Element: // Element is fluid
+        r = _test[_i][_qp] * _D[_qp] * _neighbor_dotdot[_qp] * _normals[_qp](2);
+        break;
+
+      case Moose::Neighbor: // Neighbor is structure
+        r = _test_neighbor[_i][_qp] * -_u[_qp] * _normals[_qp](2);
+        break;
+    }
+  }
+
+  return r;
+}
+
+Real
+FluidStructureInterface::computeQpJacobian(Moose::DGJacobianType type)
+{
+
+  Real jac = 0;
+
+  if (_component == 0)
```

Comment:
switch

Reply:
should still be a nested switch. You added that as commented out code and removed it again